### DISCUSSION TITLE
쿠키와 세션

### DISCUSSION THREAD

**Discussion by yunjjeongjo** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#62] [Interview Question]
Q1) 쿠키와 세션에 대해서 설명해주시고 이 둘의 차이점을 설명해주세요. (저장 위치, lifecycle, 속도 등을 비교하며 설명해 주시면 좋습니다)

**Comment by punchdrunkard** [comment]
쿠키와 세션은 http의 stateless 특성을 보완하기 위해 사용합니다. 

쿠키는 클라이언트 측에 저장되고, 세션 쿠키와 지속 쿠키로 나뉩니다. 세션 쿠키의 경우, 브라우저가 닫히면 사라집니다. 만료 시간을 지정하지 않는다면 쿠키는 기본적으로 세션 쿠키로 지정됩니다. 지속 쿠키는 반대로, 브라우저가 닫히면 사라지지 않고, 만료 시점에 지워집니다.

세션의 경우, 서버 측에 저장되는데 보통 세션에 접근하기 위하여 세션 key를 브라우저 쿠키에 저장하고, 트랜잭션에서 이를 첨부하여 세션의 정보를 가져올 수 있습니다.

이들의 속도는 상대적으로 브라우저에서 바로 가져올 수 있는 쿠키가 빠른 편입니다.
